### ISSUE TITLE
Review Poller.PollUntilDone method

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-sdk-for-go#17653]
[PollUntilDone](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/poller.go#L134) takes a context and a `time.Duration` parameter that's used as the sleep interval between polls when there's no `Retry-After` header.  I have two concerns with this method.

1. It's not clear what value to use for `freq`.
2. There is no options bag to accommodate future expansion.

For the first bullet, we have a few options.

- Doc it (this is the minimum IMO)
- Pass zero to accept a default

If we accept zero to accept a default, we at least know if we're creating a poller for control vs data-plane, which have pretty different default values (30s for the former, the latter is in order of seconds).

We could also consider expanding [NewPollerOptions](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/poller.go#L42) to take a default polling interval at poller construction, which would allow services to set their own default polling interval.

The second bullet I'm less concerned about.  However, we should ensure that we're all ok with it.

**Comment by JeffreyRichter** [comment]
I'm fine with PollUntilDone taking a pointer to an options struct that (today) has just 1 freq field in it; nil would mean default.
Perhaps for ARM, you could set a default (30s). But for data-plane, I really struggle with allowing a default because it is so different across operations. In fact, even the same operation (like copy blob) could have a short delay for a small blob and a large delay for a big blob. Only the customer knows if the blob is small or large; there is no way the SDK could know. Furthermore, Storage charges real money for each I/O operation so polling frequently can get expensive quickly - customers must be in control of this to avoid any surprising bills from Azure. 

So, I actually think that mandating a frequency is fine and we could/should document some guidelines (as you point out). Also, I think mandating for data plane and having a default for control plane is odd too due to the inconsistency and customers would really have to understand when they are polling on a data plane vs control plane operation and this may not be so obvious.

So, in summary, I think the simple/consistent thing is best: We can document that the default is consistently 30s for ALL pollers. So, 0/nil means 30 seconds. And, we also document this clearly and state that this works well for many Azure services/operations but that customers should really consider passing a more appropriate value here based on latency, network I/Os and costs.

**Comment by jhendrixMSFT** [comment]
Fixed in upcoming `azcore@v0.24.0`
